src/js/components/FocusedContainer.js

```diff
@@ -18,44 +13,51 @@ export const FocusedContainer = ({
   const [bodyOverflowStyle, setBodyOverflowStyle] = useState('');
   const ref = useRef(null);
 
+  const root = useContext(RootsContext);
+  const [updatedRoot, setUpdatedRoot] = useState(root);
+
   useEffect(() => {
-    const removeTrap = () => {
-      const child = ref.current;
-      getBodyChildElements()
-        .filter(isNotAncestorOf(child))
-        .forEach(makeNodeFocusable);
-      if (restrictScroll) {
-        document.body.style.overflow = bodyOverflowStyle;
-      }
-    };
+    // make sure value of null is not added to array
+    if (ref.current) setUpdatedRoot([...root, ref.current]);
+  }, [root]);
 
-    const handleTrapFocus = () => {
-      const child = ref.current;
-      getBodyChildElements()
-        .filter(isNotAncestorOf(child))
-        .forEach(makeNodeUnfocusable);
+  useEffect(() => {
+    if (
+      bodyOverflowStyle !== 'hidden' &&
+      !hidden &&
+      restrictScroll &&
+      trapFocus
+    ) {
+      setBodyOverflowStyle(document.body.style.overflow);
+      document.body.style.overflow = 'hidden';
+    }
 
-      if (restrictScroll && bodyOverflowStyle !== 'hidden') {
-        setBodyOverflowStyle(document.body.style.overflow);
-        document.body.style.overflow = 'hidden';
+    return () => {
+      if (restrictScroll) {
```

Comment:
It feels like this test should align more with the test at lines 26-28.

Reply:
That makes sense. I've updated to condition.

Reply:
Done.